include/zephyr/dt-bindings/reserved-memory/nordic-owned-memory.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_RESERVED_MEMORY_NORDIC_OWNED_MEMORY_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_RESERVED_MEMORY_NORDIC_OWNED_MEMORY_H_
+
+#include <zephyr/dt-bindings/dt-util.h>
+
+/**
+ * @name Basic memory permission flags.
+ * @{
+ */
+
+/** Readable. */
+#define NRF_PERM_R BIT(0)
+/** Writable. */
+#define NRF_PERM_W BIT(1)
+/** Executable. */
+#define NRF_PERM_X BIT(2)
+/** Secure-only. */
+#define NRF_PERM_S BIT(3)
+/** Non-secure-callable. */
+#define NRF_PERM_NSC BIT(4)
+
+/**
+ * @}
+ */
+
+/**
+ * @name Memory permission flag combinations.
+ * @note NRF_PERM_NSC overrides all other flags, so it is not included here.
+ * @{
+ */
+
+#define NRF_PERM_RW   (NRF_PERM_R | NRF_PERM_W)
+#define NRF_PERM_RX   (NRF_PERM_R | NRF_PERM_X)
+#define NRF_PERM_RS   (NRF_PERM_R | NRF_PERM_S)
+#define NRF_PERM_WX   (NRF_PERM_W | NRF_PERM_X)
```

Comment:
Seems like it shouldn't be, unless it's technically allowed by the bus architecture. I'm not familiar enough with it.

Either way, it was also possible to set `perm-execute` without `perm-read` with the old binding, so I'm not sure we need to add new restrictions here.

Reply:
I'm pretty sure the bus allows for this, but I can't imagine a master that would send a transaction with an X attribute but without R.

My comment was intended to suggest removing the `define`s that won't be ever used. Just to avoid maintaining dead code. But I'm fine to keep it.